### PR TITLE
updated keymint provision instruction set to allign with keymaster in…

### PR DESCRIPTION
**Pull request by AvinashHedage**
…struction set

### CODE REVIEW

File: Applet/AndroidSEProvider/src/com/android/javacard/keymaster/KMAndroidSEApplet.java [hunk 1]
```diff
@@ -43,21 +43,19 @@ public class KMAndroidSEApplet extends KMKeymasterApplet implements OnUpgradeLis
...
+  private static final byte INS_GET_PROVISION_STATUS_CMD = INS_KEYMINT_PROVIDER_APDU_START + 7;
+  private static final byte INS_SE_FACTORY_PROVISIONING_LOCK_CMD = INS_KEYMINT_PROVIDER_APDU_START + 10;
+  private static final byte INS_PROVISION_OEM_ROOT_PUBLIC_KEY_CMD = INS_KEYMINT_PROVIDER_APDU_START + 11;
+  private static final byte INS_OEM_UNLOCK_PROVISIONING_CMD = INS_KEYMINT_PROVIDER_APDU_START + 12;  
   private static final byte INS_PROVISION_RKP_DEVICE_UNIQUE_KEYPAIR_CMD =
-      INS_KEYMINT_PROVIDER_APDU_START + 6;
+      INS_KEYMINT_PROVIDER_APDU_START + 13;
   private static final byte INS_PROVISION_RKP_ADDITIONAL_CERT_CHAIN_CMD =
-      INS_KEYMINT_PROVIDER_APDU_START + 7;
-  private static final byte INS_SET_BOOT_ENDED_CMD =
```
Comment (subrahmanyaman) [inline Applet/AndroidSEProvider/src/com/android/javacard/keymaster/KMAndroidSEApplet.java:56]: Write a comment that INS_SET_BOOT_ENDED_CMD was reserved on 0x09 earlier. Same in KMJcardSimApplet.

File: Applet/AndroidSEProvider/src/com/android/javacard/keymaster/KMAndroidSEApplet.java [hunk 2]
```diff
@@ -43,21 +43,19 @@ public class KMAndroidSEApplet extends KMKeymasterApplet implements OnUpgradeLis
...
-  private static final byte INS_PROVISION_ATTEST_IDS_CMD = INS_KEYMINT_PROVIDER_APDU_START + 1;
+  private static final byte INS_PROVISION_ATTEST_IDS_CMD = INS_KEYMINT_PROVIDER_APDU_START + 3;
   private static final byte INS_PROVISION_PRESHARED_SECRET_CMD =
-      INS_KEYMINT_PROVIDER_APDU_START + 2;
-  private static final byte INS_OEM_LOCK_PROVISIONING_CMD = INS_KEYMINT_PROVIDER_APDU_START + 3;
-  private static final byte INS_GET_PROVISION_STATUS_CMD = INS_KEYMINT_PROVIDER_APDU_START + 4;
+      INS_KEYMINT_PROVIDER_APDU_START + 4;
   private static final byte INS_SET_BOOT_PARAMS_CMD = INS_KEYMINT_PROVIDER_APDU_START + 5;
+  private static final byte INS_OEM_LOCK_PROVISIONING_CMD = INS_KEYMINT_PROVIDER_APDU_START + 6;
+  private static final byte INS_GET_PROVISION_STATUS_CMD = INS_KEYMINT_PROVIDER_APDU_START + 7;
```
Comment (subrahmanyaman) [inline Applet/AndroidSEProvider/src/com/android/javacard/keymaster/KMAndroidSEApplet.java:51]: Keymaster uses INS_SET_VERSION_PATCHLEVEL_CMD(8) name to set osversion, patchLevel and vendorPatchLevel, where as KeyMint uses INS_INIT_STRONGBOX_CMD(58) name to set osversion, patchLevel and vendorPatchLevel. we need to make INS_INIT_STRONGBOX_CMD to 8 in KeyMint.

File: ProvisioningTool/include/constants.h [hunk 3]
```diff
@@ -101,14 +101,16 @@ constexpr char kSeFactoryProvisionLock[] = "se_factory_lock";
 constexpr char kUnLockProvision[] = "unlock_provision";
 
 // Instruction constatnts
-// TODO Modify according to keymint
-constexpr int kAttestationIdsCmd = INS_BEGIN_KM_CMD + 1;
-constexpr int kPresharedSecretCmd = INS_BEGIN_KM_CMD + 2;
-constexpr int kOemLockProvisionCmd = INS_BEGIN_KM_CMD + 3;
```
Comment (subrahmanyaman) [inline ProvisioningTool/include/constants.h:107]: Looks like the HAL changes are missing.

**Review by subrahmanyaman — COMMENTED** [review]
Looks like the HAL changes are missing.

File: Applet/AndroidSEProvider/src/com/android/javacard/keymaster/KMAndroidSEApplet.java (hunk 1 above)
Comment (AvinashHedage) [reply Applet/AndroidSEProvider/src/com/android/javacard/keymaster/KMAndroidSEApplet.java:56]: Done

File: ProvisioningTool/include/constants.h (hunk 3 above)
Comment (AvinashHedage) [reply ProvisioningTool/include/constants.h:107]: Made HAL changes wrt init strong box command

File: Applet/AndroidSEProvider/src/com/android/javacard/keymaster/KMAndroidSEApplet.java (hunk 2 above)
Comment (AvinashHedage) [reply Applet/AndroidSEProvider/src/com/android/javacard/keymaster/KMAndroidSEApplet.java:51]: Done

**Pull request closed without merging**